src/AnalysisPrograms/Recognizers/Birds/AnthusNovaeseelandiae.cs

```diff
@@ -0,0 +1,281 @@
+        /// The constants for this method were obtained from the calls in a Gympie recording obtained by Yvonne Phillips.
+        /// </summary>
+        /// <param name="ev">An event containing at least one forward track i.e. a chirp.</param>
+        public static void SetFrequencyProfileScore(ChirpEvent ev)
+        {
+            const double risingDuration = 0.06;
+            const double gapDuration = 0.03;
+            const double fallingDuration = 0.06;
+
+            var track = ev.Tracks.First();
+            var profile = track.GetTrackFrequencyProfile().ToArray();
+
+            // get the first point
+            var firstPoint = track.Points.First();
+            var frameDuration = firstPoint.Seconds.Maximum - firstPoint.Seconds.Minimum;
+            var risingFrameCount = (int)Math.Floor(risingDuration / frameDuration);
+            var gapFrameCount = (int)Math.Floor(gapDuration / frameDuration);
+            var fallingFrameCount = (int)Math.Floor(fallingDuration / frameDuration);
+
+            var startSum = 0.0;
+            if (profile.Length >= risingFrameCount)
+            {
+                for (var i = 0; i <= risingFrameCount; i++)
+                {
+                    startSum += profile[i];
+                }
+            }
+
+            int startFrame = risingFrameCount + gapFrameCount;
+            int endFrame = startFrame + fallingFrameCount;
+            var endSum = 0.0;
+            if (profile.Length >= endFrame)
+            {
+                for (var i = startFrame; i <= endFrame; i++)
+                {
+                    endSum += profile[i];
+                }
+            }
+
+            // set score to 1.0 if the profile has inverted U shape.
+            double score = 0.0;
+            if (startSum > 0.0 && endSum < 0.0)
+            {
+                score = 1.0;
+            }
+
+            ev.FrequencyProfileScore = score;
+        }
+
+        /// <summary>
+        /// .
+        /// </summary>
+        /// <param name="points">List of spectral points.</param>
+        public static void WriteFrequencyProfile(ISpectralPoint[] points)
+        {
+            /* Here are the frequency profiles of some events.
+             * Note that the first five frames (0.057 seconds) have positive slope and subsequent frames have negative slope.
+             * The final frames are likely to be echo and to be avoided.
+             * Therefore take the first 0.6s to calculate the positive slope, leave a gap of 0.025 seconds and then get negative slope from the next 0.6 seconds.
+            */
+
+            if (points != null)
+            {
+                var str = $"Track({points[0].Seconds.Minimum:F2}):";
+
+                foreach (var point in points)
+                {
+                    str += $" {point.Hertz.Minimum},";
+                }
+
+                Console.WriteLine(str);
+            }
```

Comment:
All of this code seems like it should be factored out. Two of these methods are copy and pasted from the NinoxBoobook class!

Reply:
OK but I was thinking to refactor them later. Rather than have to find them again.